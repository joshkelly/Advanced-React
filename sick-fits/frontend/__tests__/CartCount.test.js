import { render, screen } from '@testing-library/react';
import wait from 'waait';
import CartCount from '../components/CartCount';

describe('<CartCount />', () => {
  it('Renders', () => {
    render(<CartCount count={10} />);
  });

  it('Matches snapshot', () => {
    const { container, debug } = render(<CartCount count={15} />);
    expect(container).toMatchSnapshot();
  });

  it('updates via props', async () => {
    const { container, rerender, debug } = render(<CartCount count={15} />);

    expect(container.textContent).toBe('15');
    // expect(container).toHaveTextContent('15'); Same as above just a style preference.

    // Update the props
    rerender(<CartCount count={10} />);

    // wait for X ms
    await wait(400);
    // await screen.findByText('10'); - Doesn't work, tests 1015 as found 10.

    expect(container.textContent).toBe('10');
    expect(container).toMatchSnapshot();
  });
});
