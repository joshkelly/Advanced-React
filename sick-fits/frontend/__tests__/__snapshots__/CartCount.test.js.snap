// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> Matches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 gwXiGr"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 GxMDG count"
      >
        15
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount /> updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-xxvp4g-1 gwXiGr"
  >
    <div>
      <div
        class="CartCount__Dot-sc-xxvp4g-0 GxMDG count count-enter-done"
      >
        10
      </div>
    </div>
  </span>
</div>
`;
